### ISSUE TITLE
Does pytest-doctestplus support testcode and testoutput directives?

### ISSUE THREAD

**Issue by ArtyomKaltovich** [astropy/pytest-doctestplus#106]
Hello.

I've just install pytest-doctestplus, so sorry, for probably stupid question.

I am written the following doc:

```
>>> 43 + 1
44

.. testcode::
    43 + 1

The result is:

.. testoutput::
    44
```

But when I run pytest only first test is executed.

While is is also correct way for defining tests: https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html

Regards.

**Comment by bsipocz** [comment]
Please note that sphinx doctest extension is not the same as the pytest-doctestplus plugin. In your exampe, you need to add `>>>` to the testcode as well to be picked up by pytest.

**Comment by ArtyomKaltovich** [comment]
Hello @bsipocz 

But do you have plans to support it?

Maybe you know if it is difficult to add? :) I can try if it doesn't take much time.

**Comment by TiemenSch** [comment]
What would need to happen for this extension to pickup the ``.. testcode::`` and ``.. testoutput::`` directives?

As they are pretty much introduced in the same place in Sphinx's docs (https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html), I think it would be rather intuitive to include them with this extension for pytest, too.

Copying code snippets from ``.. testcode::`` blocks is way easier for users reading documentation than the ``.. doctest::`` ones with the ``>>>`` littered everywhere IMHO.

**Comment by pllim** [comment]
I don't think we have plans to add this support, as we don't use it ourselves in the Astropy project. You can copy the `>>>` directly into IPython session and the session is smart enough to ignore the `>>>`, so we have gotten no such complains from users (yet).

**Comment by TiemenSch** [comment]
Ah, that makes sense. I understood this was separated from the rest of the codebase to be a bit more standalone applicable for "any" pytest/Python user.

For non-IPython console users (e.g. flat Python files/packages or when working in JupyterLab/notebooks), this is a bit of a hassle.

**Comment by pllim** [comment]
I see a similar discussion upstream at readthedocs/sphinx_rtd_theme#167 . I did remember a long time ago I came across a doc that lets you toggle the `>>>` on/off but now I can't find it anymore.

**Comment by saimn** [comment]
Probably https://github.com/astropy/astropy-sphinx-theme/blob/master/astropy_sphinx_theme/bootstrap-astropy/static/copybutton.js ;-)

**Comment by pllim** [comment]
Ah, yes. It is still there. I was blind. 😆 

![Screenshot 2021-03-03 100708](https://user-images.githubusercontent.com/2090236/109826056-6cb02d80-7c08-11eb-927d-1117609f9207.jpg)

**Comment by TiemenSch** [comment]
That https://github.com/jurasofish/sphinx-toggleprompt extension is a nice workaround to be able to get the ``>>>`` out of the way quickly :)

I think there is quite some overlap with the logic that's already implemented over at https://github.com/thisch/pytest-sphinx 
That seems to be a self-made plugin to support more or less the same as this extension does, except for the other directives. A little less mature and not fully supporting the directive options, though.

Thanks for the quick responses btw!

**Comment by lpsinger** [comment]
Here's a little Sphinx extension that monkeypatches pytest-doctestplus and sphinx.ext.doctest to at least get `FLOAT_CMP` working:

```python
def setup(*args, **kwargs):
    import doctest
    import pytest_doctestplus.output_checker
    import sphinx.ext.doctest

    doctest.OutputChecker = pytest_doctestplus.output_checker.OutputChecker
    return sphinx.ext.doctest.setup(*args, **kwargs)
```
